native/internal/math32/sqrt.go

```diff
@@ -0,0 +1,24 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//+build !amd64 noasm
+
+package math32
+
+import (
+	"math"
+)
+
+// Sqrt returns the square root of x.
+//
+// Special cases are:
+//	Sqrt(+Inf) = +Inf
+//	Sqrt(±0) = ±0
+//	Sqrt(x < 0) = NaN
+//	Sqrt(NaN) = NaN
+func Sqrt(x float32) float32 {
+	// FIXME(kortschak): Direct translation of the math
```

Comment:
While the comment originally refered to the Go implementation, it is still now true for the 386 and arm asm.

I tried the translation of the 386 asm and I think that the single precision instructions are not registered. I don't have hardware to check the arm code.

The Go sqrt code is used in this package as it is in "math", as a reference implementation.

I'll add to the comment that this is refering to the asm.